dome9/resource_dome9_cloudaccount_kubernetes.go

```diff
@@ -138,3 +199,203 @@ func createKubernetesCloudAccountRequest(d *schema.ResourceData) k8s.CloudAccoun
 		OrganizationalUnitID: d.Get("organizational_unit_id").(string),
 	}
 }
+
+func featuresCreate(d *schema.ResourceData, d9Client *Client, newId string) error {
+	runtimeProtection, ok := d.GetOk("runtime_protection")
+	if ok {
+		if err := configureRuntimeProtection(runtimeProtection, newId, d9Client); err != nil {
+			return err
+		}
+	}
+
+	admissionControl, ok := d.GetOk("admission_control")
+	if ok {
+		if err := configureAdmissionControl(admissionControl, newId, d9Client); err != nil {
+			return err
+		}
+	}
+
+	imageAssurance, ok := d.GetOk("image_assurance")
+	if ok {
+		if err := configureImageAssurance(imageAssurance, newId, d9Client); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func featuresUpdate(d *schema.ResourceData, d9Client *Client) error {
+	if d.HasChange("runtime_protection") {
+		log.Println("Runtime Protection has been changed")
+
+		runtimeProtection := d.Get("runtime_protection")
```

Comment:
Good point, I think I missed this sort of test - will check and fix as needed

Reply:
Update following a check:
It seems that if you remove a feature block HasChanged() for it is False.
So, for example, if you had all feature blocks and removed them, 'terraform plan' will indicate no required changes.
I think this is since the feature blocks are 'Optional'.
setup-wise, if you remove such a block the setup will remain with the configuration it had for it

Reply:
Thanks,
Sounds like a limitation in TF that we are not supposed to cover, right?

Reply:
As far as I understand that's the expected behavior for this kind of usage.
Optional - so taken from configuration if exists.
but also Computed - so taken from known state if not in configuration.
(See somewhat relevant discussion: https://discuss.hashicorp.com/t/providers-schema-how-to-distinguish-attributes-from-arguments/5029/3)